frontend/app/src/components/home/groupdetails/Members.svelte

```diff
@@ -1,37 +1,73 @@
 <script lang="ts">
+    import { _ } from "svelte-i18n";
     import Search from "../../Search.svelte";
     import Member from "./Member.svelte";
+    import BlockedUser from "./BlockedUser.svelte";
     import MembersHeader from "./MembersHeader.svelte";
     import VirtualList from "../../VirtualList.svelte";
     import type {
-        BlockedMember,
         FullMember,
         GroupChatSummary,
         Member as MemberType,
         OpenChat,
+        PartialUserSummary,
         UserLookup,
     } from "openchat-client";
-    import { createEventDispatcher, getContext } from "svelte";
-    import type { Readable } from "svelte/store";
+    import { createEventDispatcher, getContext, onMount } from "svelte";
+    import MembersSelectionButton from "../MembersSelectionButton.svelte";
+    import InvitedUser from "./InvitedUser.svelte";
+    import ViewUserProfile from "../profile/ViewUserProfile.svelte";
 
     const client = getContext<OpenChat>("client");
     const userId = client.user.userId;
 
     export let closeIcon: "close" | "back";
-    export let members: Readable<MemberType[]>;
-    export let blockedUsers: Readable<Set<string>>;
     export let chat: GroupChatSummary;
 
+    $: members = client.currentChatMembers;
+    $: blocked = client.currentChatBlockedUsers;
+    $: invited = client.currentChatInvitedUsers;
     $: userStore = client.userStore;
-    $: knownUsers = getKnownUsers($userStore, $members, $blockedUsers);
+    $: knownUsers = getKnownUsers($userStore, $members);
     $: me = knownUsers.find((u) => u.userId === userId);
-    $: others = knownUsers
-        .filter((u) => matchesSearch(searchTerm, u) && u.userId !== userId)
+    $: fullMembers = knownUsers
+        .filter(
+            (u) =>
+                matchesSearch(searchTerm, u) &&
+                u.userId !== userId &&
+                !$blocked.has(u.userId) &&
+                !$invited.has(u.userId)
```

Comment:
@julianjelfs I don't think this is necessary is it? "currentChatMembers" won't contain blocked or invited.
```
    !$blocked.has(u.userId) &&
    !$invited.has(u.userId)
```

Reply:
Quite possibly not necessary. I had to stop kind of in the middle so I'm not really sure.